app/controllers/users_controller.rb

```diff
@@ -0,0 +1,51 @@
+class UsersController < ApplicationController
+  enable :sessions
+  set :session_secret, 'fwitter'
+  set :public_folder, 'public'
+  set :views, 'app/views'
+
+# Displayse the "Sign Up" form and creates a new user
+  get '/signup' do
+    redirect '/tweets' if logged_in?
+    erb :'users/create'
+  end
+
+  post '/signup' do
+    redirect '/signup' if params['username'] == "" || params['email'] == "" || params['password'] == ""
```

Comment:
I think you could use [ActiveRecord validations](https://learn.co/tracks/full-stack-web-development/sinatra/activerecord/user-authentication-in-sinatra) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.
Also, since the params hash keys are named the same as the actual attributes, you can get away with just running `user = User.create(params)` or if incorporating Active Record validations: `user = User.new(params)`